salt/transport/zeromq.py

```diff
@@ -13,6 +13,7 @@
 import hashlib
 import logging
 import weakref
+import threading
```

Comment:
We're not creating any threads here. This import is for thread local storage `threading.local`. Since the zmq socket is not thread save, we're caching it in thread local storage rather than just on the `ZeroMQPubServerChannel`. It's safer than storing it on class and facilitates testing (where we use threads). Under normal operation, there will only be one socket per MWorker because the publish method is called from the same thread every time.

As for async, I think we should move this to zmq's async. I figured that should happen in develop after this get's merged up.